DataFormats/FWLite/src/DataGetterHelper.cc

```diff
@@ -131,20 +133,22 @@ namespace fwlite {
         //obj.destruct();
         ////END OF WORK AROUND
 
-        TTreeCache* tcache = dynamic_cast<TTreeCache*> (branchMap_->getFile()->GetCacheRead());
+        if (tcUse_) {
+            TTreeCache* tcache = dynamic_cast<TTreeCache*> (branchMap_->getFile()->GetCacheRead());
 
-        if (nullptr == tcache) {
-            iData.branch_->GetEntry(eventEntry);
-        } else {
-            if (!tcTrained_) {
-                tcache->SetLearnEntries(100);
-                tcache->SetEntryRange(0, tree_->GetEntries());
-                tcTrained_ = true;
+            if (nullptr != tcache) {
+                if (!tcTrained_) {
+                    tcache->SetLearnEntries(100);
+                    tcache->SetEntryRange(0, tree_->GetEntries());
+                    tcTrained_ = true;
+                }
+                tree_->LoadTree(eventEntry);
             }
-            tree_->LoadTree(eventEntry);
-            iData.branch_->GetEntry(eventEntry);
-       }
-       iData.lastProduct_=eventEntry;
+        }
+        branchAccFoo_(iData.branch_);
```

Comment:
I had mis-remembered the relationship between DataGetterHelper and Event. 
I am not comfortable with the use of std::functional in this context since it is to easy for abuse. I'm still considering how to it could be accomplished in a more 'controlled' way.